java/ql/lib/semmle/code/java/security/TaintedEnvironmentVariableQuery.qll

```diff
@@ -0,0 +1,53 @@
+/** Modules to reason about the tainting of environment variables */
+
+private import semmle.code.java.dataflow.ExternalFlow
+private import semmle.code.java.dataflow.FlowSources
+private import semmle.code.java.Maps
+private import semmle.code.java.JDK
+
+private class MapPutOrRemove extends MethodCall {
+  MapPutOrRemove() {
+    this.getMethod() instanceof MapMutator and
+    this.getMethod().getName().matches(["put%", "remove"])
```

Comment:
There does not appear to be a test for `putAll`?

Also, what about [putIfAbsent](https://download.java.net/java/early_access/panama/docs/api/java.base/java/util/Map.html#putIfAbsent(K,V))?
This inherently has the potential for FPs.

If this is acceptable, why do we not model [replace](https://download.java.net/java/early_access/panama/docs/api/java.base/java/util/Map.html#replace(K,V))?
It should have the same inherent potential for FPs.

If one wants to get fancy, they could also add support for the `compute(IfAbsent|IfPresent)` methods although they are probably rarely used in practice.